consensus/consensus.go

```diff
@@ -254,6 +243,59 @@ func New(host p2p.Host, ShardID uint32, leader p2p.Peer, blsPriKey *bls.SecretKe
 	// channel for receiving newly generated VDF
 	consensus.RndChannel = make(chan [vdfAndSeedSize]byte)
 	consensus.uniqueIDInstance = utils.GetUniqueValidatorIDInstance()
-	memprofiling.GetMemProfiling().Add("consensus.pbftLog", consensus.PbftLog)
+	memprofiling.GetMemProfiling().Add("consensus.pbftLog", consensus.PBFTLog)
 	return &consensus, nil
 }
+
+// NewStakedVotePerValidator makes a staking based consensus object
+func NewStakedVotePerValidator(
+	host p2p.Host, shard uint32, leader p2p.Peer, blsPriKey *bls.SecretKey,
+) (*Consensus, error) {
+	c, oops := baseConsensus(host, shard, leader, blsPriKey)
+	if oops != nil {
+		return nil, oops
+	}
+	c.ConsensusMechanism = values.SuperMajorityStake
+	c.isQuorumSatisfied = func(p values.QuorumPhase) bool {
+		return true
+	}
+	c.QuorumThreshold = func() *big.Int {
+		return big.NewInt(0)
+	}
+	c.rewardThreshold = func() *big.Int {
+		return big.NewInt(int64(len(c.PublicKeys) * 9 / 10))
+	}
+	return c, nil
+}
+
+// TODO: put shardId into chain reader's chain config
+
+// NewOneVotePerValidator is the original, pre-PoS based consensus object
+func NewOneVotePerValidator(
+	host p2p.Host, shard uint32, leader p2p.Peer, blsPriKey *bls.SecretKey,
+) (*Consensus, error) {
+	c, oops := baseConsensus(host, shard, leader, blsPriKey)
+	if oops != nil {
+		return nil, oops
+	}
+	c.ConsensusMechanism = values.SuperMajorityVote
+	c.isQuorumSatisfied = func(p values.QuorumPhase) bool {
+		need := c.QuorumThreshold().Int64()
+		switch p {
+		case values.QuorumPrepare:
+			return int64(len(c.prepareSigs)) >= need
+		case values.QuorumCommit:
+			return int64(len(c.commitSigs)) >= need
+		default:
+			// Not possible
+			return false
+		}
+	}
+	c.QuorumThreshold = func() *big.Int {
+		return big.NewInt(int64(len(c.PublicKeys)*2/3 + 1))
+	}
+	c.rewardThreshold = func() *big.Int {
+		return big.NewInt(int64(len(c.PublicKeys) * 9 / 10))
+	}
+	return c, nil
+}
```

Comment:
Please avoid introducing multiple variant-specific factory methods, but consider factoring varied logic into separate implementation of a common interface then use DI idiom:

```go
type Quorum interface {
	Mechanism() values.ConsensusMechanism
	IsSatisfied(sigs, constituents int) bool
	QuorumThreshold(constituents int) int
	RewardThreshold(constituents int) int
}

type VoteBasedQuorum struct {}

type StakeBasedQuorum struct {}
```

With a few more steps you should also be able to eliminate `sigs` and `constituents` arguments from methods, by injecting accessors of the number of signatures/constituents in the current context into the quorum check implementations.